### PR TITLE
Fix and adjust the template content element/module

### PR DESCRIPTION
**Pull request by m-vo**
As discussed in the last Contao call, this PR adds a Twig template for the `template` element/module. I also made sure the two controllers were in line again and added a note regarding the existing fragment controllers to the `UPGRADE.md`.

### CODE REVIEW

**Review by fritzmg — DISMISSED** [review]

File: core-bundle/src/Controller/ContentElement/TemplateController.php [hunk 1]
```diff
@@ -14,25 +14,28 @@
...
+            array_combine(
+                array_column($data, 'key'),
+                array_column($data, 'value')
+            )
         );
 
-        // Backwards compatibililty
-        $template->data = $data;
+        // Backwards compatibility
+        $template->set('data', $data);
```
Comment (ausi) [inline core-bundle/src/Controller/ContentElement/TemplateController.php]: Do we still need this backwards compatibility?

**Review by ausi — DISMISSED** [review]

File: core-bundle/src/Controller/ContentElement/TemplateController.php (hunk 1 above)
Comment (m-vo) [reply core-bundle/src/Controller/ContentElement/TemplateController.php]: I wasn't sure about this tbh. IIRC we intentionally left it in for 5.0 but then again we did not think about making the templates Twig only back then. Shall we remove it?

Comment (ausi) [reply core-bundle/src/Controller/ContentElement/TemplateController.php]: I’d remove it as the templates are not backwards compatible anyways.

File: core-bundle/src/Controller/ContentElement/TemplateController.php [hunk 2]
```diff
@@ -14,25 +14,28 @@
...
 
         return $template->getResponse();
```
Comment (m-vo) [inline core-bundle/src/Controller/ContentElement/TemplateController.php]:
```suggestion
        return $template->getResponse();
```

File: core-bundle/src/Controller/FrontendModule/TemplateController.php [hunk 3]
```diff
@@ -13,20 +13,29 @@
...
+        $template->set(
+            'keys',
+            array_combine(
+                array_column($data, 'key'),
+                array_column($data, 'value')
+            )
+        );
+
+        // Backwards compatibility
+        $template->set('data', $data);
```
Comment (m-vo) [inline core-bundle/src/Controller/FrontendModule/TemplateController.php]:
```suggestion

```

File: core-bundle/src/Controller/ContentElement/TemplateController.php (hunk 1 above)
Comment (m-vo) [reply core-bundle/src/Controller/ContentElement/TemplateController.php]: Done in b67226d

File: core-bundle/src/Controller/FrontendModule/TemplateController.php [hunk 4]
```diff
@@ -13,21 +13,26 @@
...
+        $data = StringUtil::deserialize($model->data, true);
 
+        $template->set(
+            'keys',
+            array_combine(
+                array_column($data, 'key'),
+                array_column($data, 'value')
+            )
+        );
         return $template->getResponse();
```
Comment (m-vo) [inline core-bundle/src/Controller/FrontendModule/TemplateController.php:37]:
```suggestion

        return $template->getResponse();
```

**Review by ausi — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @m-vo.

**Pull request merged**